### ISSUE TITLE
Bug: Infinite Scroll plus Carousel modules active, with php7.1, can raise a PHP Fatal error:  Uncaught ArgumentCountError

### ISSUE THREAD

**Issue by eri-trabiccolo** [Automattic/jetpack#7939]
Hello folks.

#### Steps to reproduce the issue
Here's a way to reproduce the issue:

with a Twenty Seventeen child-theme active

- Enable Jetpack -> Settings -> Media -> Display images and galleries in a gorgeous, full-screen browsing experience
- Enable Jetpack -> Settings -> Theme enhancements -> Infinite Scroll -> Load more posts in page with a button
- Save Settings
- In your child-theme functions.php use this test snippet:
```php
add_filter( 'post_gallery', 'post_gallery_callback', 10, 3 );
function post_gallery_callback( $output = '', $atts, $instance ) {
	return $output;
}
```

#### What I expected
No php fatal error raised :D 

#### What happened instead
`PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function post_gallery_callback(), 2 passed in /srv/www/wordpress-default/public_html/wp-includes/class-wp-hook.php on line 298 and exactly 3 expected in /srv/www/wordpress-default/public_html/wp-content/themes/twentyseventeen-child/functions.php:12`

#### Why and how to solve?
Here's the "culprit" line of code:
https://github.com/Automattic/jetpack/blob/5.4/modules/infinite-scroll.php#L223

Basically, since WP 4.4.2 there are 3 required params for the `post_gallery` filter
https://core.trac.wordpress.org/browser/trunk/src/wp-includes/media.php#L1652

I guess it can be solved by adding a third integer param (say 0).

p.s.
before php7.1 that would just generate a warning:
https://secure.php.net/manual/en/migration71.incompatible.php
